### DISCUSSION TITLE
try to use SES in Soto SDK with Vapor 4 but I have some error

### DISCUSSION THREAD

**Discussion by NerisD** [soto-project/soto#709] [Q&A]
**Describe the bug**
I want to use SES Services with Vapor 4 to send an Email with my own route. 

I follow step by step the recommendation in Soto Doc to use Vapor 4. 

But I got this error : 
AsyncHTTPClient/HTTPClient.swift:591: Precondition failed: Provided EventLoop must be part of clients EventLoopGroup.

Some one can give me some explanations or explain how can I implement Soto SES with Vapor. 

Thanks :)

**Comment by adam-fowler** [comment]
Are you creating your `AWSClient` with `httpClientProvider` set to `.createNew` and when calling SES methods setting `eventLoo`p to `req.eventLoop`? If so you are asking the HTTP client to run on an `EventLoop` from a different `EventLoopGroup`. 

If you want to use the request event loop you need to use an HTTP client that uses the same EventLoopGroup. Luckily Vapor provides one. I believe you can get it via `app.client.shared`. You should then create your `AWSClient` as follows

```swift
let awsClient = AWSClient(httpClientProvider: .shared(app.client.shared))
```

**Reply by NerisD** [reply]
Oh my god ! 

Thanks ! I think this problem solve. 

I used .createNew instead of use .shared(app.http.client.shared)

Now I need to understand how I can change my region to match with my credential. Actually it use the default one ( US-EAST-1) but mine is eu-west-3 

Thanks !

**Comment by adam-fowler** [comment]
Converting this to a discussion, as it is not an issue
